extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/java/org/apache/guacamole/auth/jdbc/base/ModeledDirectoryObjectService.java

```diff
@@ -429,14 +429,19 @@ public InternalType retrieveObject(ModeledAuthenticatedUser user,
         
         // Build list of implicit permissions
         Collection<ObjectPermissionModel> implicitPermissions =
-                new ArrayList<ObjectPermissionModel>(IMPLICIT_OBJECT_PERMISSIONS.length);
+                new ArrayList<>();
 
         UserModel userModel = user.getUser().getModel();
         for (ObjectPermission.Type permission : IMPLICIT_OBJECT_PERMISSIONS) {
 
+            Integer entityId = userModel.getEntityID();
+            
+            if (entityId == null)
+                continue;
```

Comment:
As far as I know, the only circumstances where `entityId` is null is where the system user is doing the work rather than an actual user.  That said, it should always be null, so I can definitely re-arrange this a bit.  I just wasn't sure where the permission comes from that grants the new entity `READ` access to itself...

Reply:
Okay, I've re-arranged this a bit - let me know if that looks okay.

Reply:
Okay, took another couple of tries, but I've tweaked this a bit more.  Let me know if that looks okay - essentially, in the `super` version of the `getImplicitPermissions()` method I'm either returning an `emptyList()` if the entity id is null, or an `unmodifiableCollection()` of permissions.  In the `UserService` version of `getImplicitPermissions()` I have it making a copy and then adding anything to it (`READ` for self) that it needs, and return another `unmodifiableCollection()` of those.  This makes the behavior consistent across both of those methods.